### PR TITLE
VIH-7543 support multiple providers

### PR DESCRIPTION
**Pull request by levibutler**
### JIRA link (if applicable) ###
VIH-7543

### Change description ###
support multiple providers

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+    public class EJudiciaryScheme : ProviderSchemeBase, IProviderSchemes
+    {
+        public EJudiciaryScheme(string eventhubPath): base(eventhubPath)
+        {
+        }
+
+        public override AuthProvider Provider => AuthProvider.EJudiciary;
+
+        public bool BelongsToScheme(JwtSecurityToken jwtSecurityToken) => jwtSecurityToken.Issuer.Contains("0b90379d-18de-426a-ae94-7f62441231e0", StringComparison.InvariantCultureIgnoreCase);
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs:17]: would it be safer to check the issuer from a config?

File: VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        protected override void SetJwtBearerOptions(JwtBearerOptions options)
+        {
+            options.Authority = "https://login.microsoftonline.com/0b90379d-18de-426a-ae94-7f62441231e0/v2.0";
+            options.TokenValidationParameters.ValidateLifetime = true;
+            options.TokenValidationParameters.NameClaimType = "preferred_username";
+            options.Audience = "a6596b93-7bd6-4363-81a4-3e6d9aa2df2b";
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs:24]: can we use config here too?

File: VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.spec.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { UserMediaDevice } from '../shared/models/user-media-device';
-import { MockLogger } from '../testing/mocks/MockLogger';
+import { MockLogger } from '../testing/mocks/mock-logger';
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.spec.ts:2]: i don't really see the need for a mock console logger, just use the actual console logger itself imo

**Review by shaed-parkar — COMMENTED** [review]
Nice work, would be worth getting one of the SDETs adding a couple of scenarios for a judge logging in with ejud accounts

File: VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs (hunk 1 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs:17]: yes :P todo

File: VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs (hunk 2 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/AuthenticationSchemes/EJudiciaryScheme.cs:24]: yes :P todo

File: VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.spec.ts (hunk 3 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/user-media-stream.service.spec.ts:2]: i only renamed it to match our naming convention :P

File: VideoWeb/VideoWeb.EventHub/Hub/EventHubClient.cs [hunk 4]
```diff
@@ -17,7 +16,6 @@
 
 namespace VideoWeb.EventHub.Hub
 {
-    [Authorize(Policy = "EventHubUser")]
```
[the remark this answers is not in the document]
Comment (levibutler) [reply VideoWeb/VideoWeb.EventHub/Hub/EventHubClient.cs:20]: nope, moved to policy scheme to set it dynamically per request

[the remark this answers is not in the document]
Comment (surfsquid) [reply VideoWeb/VideoWeb.EventHub/Hub/EventHubClient.cs:20]: hot.

**Review by surfsquid — APPROVED** [review]

**Review by shaed-parkar — APPROVED** [review]
I'll approve. Could we add tickets to clean up some of the concerns such as 2 testing consoles when the standard console logger would suffice

**Pull request merged**